src/components/GifItem/index.js

```diff
@@ -0,0 +1,28 @@
+import React, { Component } from 'react'
+import {
+  GifItemWrapper
+} from '../GifList/GifList.style'
+import { withRouter } from "react-router-dom";
+import FavoriteButton from '../FavoriteButton'
+
+export class GifItem extends Component {
+  render() {
+    const { gif, isFavorite, setFavorite } = this.props
+    return(
+      <GifItemWrapper key={gif.id}>
+        <FavoriteButton
+          isFavorite={isFavorite}
+          gif={gif}
+          setFavorite={setFavorite}
+        />
+        <img
```

Comment:
A [Link](https://reacttraining.com/react-router/web/api/Link) or [NavLink](https://reacttraining.com/react-router/web/api/Link) could help you here to avoid the use of the history object. Something like `<NavLink to="/route" activeClassName="something><img alt="" src />...`